File: php/src/Advertisement/Domain/Exceptions/InvalidUniqueIdentifierException.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\Domain\Exceptions;

use Demo\App\Common\Domain\DomainException;

final class InvalidUniqueIdentifierException extends DomainException
{
    public const string INVALID_ID_FORMAT_WITH_ID_MESSAGE = 'Invalid unique identifier format for ';

    private function __construct(string $message)
    {
        parent::__construct($message);
    }

    public static function withId(string $id): self
    {
        return new self(self::INVALID_ID_FORMAT_WITH_ID_MESSAGE . $id);
    }

    public function message(): string
    {
        return $this->getMessage();
    }
}

File: kotlin/src/main/kotlin/advertisement/domain/exceptions/InvalidUniqueIdentifierException.kt
package advertisement.domain.exceptions

import common.domain.DomainException

class InvalidUniqueIdentifierException private constructor(message: String) : DomainException(message) {

    companion object {
        fun withId(id: String): InvalidUniqueIdentifierException {
            return InvalidUniqueIdentifierException("Invalid unique identifier format for $id")
        }
    }
}
